src/Radicle/Daemon/HttpApi.hs

```diff
@@ -39,51 +77,71 @@ instance FromHttpApiData MachineId where
 instance ToHttpApiData MachineId where
     toUrlPiece (MachineId id) = id
 
-newtype QueryRequest = QueryRequest { expression :: Value }
-  deriving (Generic)
+newtype QueryRequestF a = QueryRequest { expression :: a }
+  deriving (Functor, Foldable, Traversable, Generic, A.ToJSON, A.FromJSON)
+
+type QueryRequest = QueryRequestF Value
 
-instance A.ToJSON QueryRequest where
+instance {-# OVERLAPPING #-} A.ToJSON QueryRequest where
     toJSON QueryRequest{..} = A.object
         [ "expression" .= valueToJson expression
         ]
-instance A.FromJSON QueryRequest where
+instance {-# OVERLAPPING #-} A.FromJSON QueryRequest where
```

Comment:
What is this overlapping with?

Reply:
https://github.com/radicle-dev/radicle/pull/586/files#diff-7a716c0d456d68b29a5cdf5fa62a6bf0R81

Which I add to be able to encode all the Req/Resp types in one go:
https://github.com/radicle-dev/radicle/pull/586/files#diff-7a716c0d456d68b29a5cdf5fa62a6bf0R70

I also tried to then encode/decode the JSON for the Req/Resp types in one go but got stuck with an overlap for `[a]`.